### DISCUSSION TITLE
Way to persist the buffers related to an activity

### DISCUSSION THREAD

**Discussion by idlip** [alphapapa/activities.el#116] [Ideas]
Right now, the activity remembers the buffers displayed in the activity and only open that 1 or 2 buffers in that tab.

But it would be a great use to save few buffers under the `activities-switch-to-buffer` list to persist when resume.

I mean to say, suppose I have 5 buffers related to `notes` activity, but I view only one main.org, with new session when I resume it only open and shows main.org, I would like to have all 5 buffers under `activities-switch-to-buffer` to persist or atleast know the name, so I can switch/activate the buffer easily.

Let me know if I have missed any discussion or issues on this additional buffer persistence.

Thank you for the package!

**Comment by alphapapa** [comment]
Hi,

This is not a bad idea.  The most challenging part would be to make the UI easy to use and understand (it should be easy to add buffers to the activity, to see the list of them, and to modify the list).

A similar idea I've been thinking about is to also load the buffers in a window's recent-buffers list (and/or the ones in its tab-line, when `tab-line-mode` is enabled).  That could solve the UI issue in a simple way.  What do you think?

**Reply by jdtsmith** [reply]
I mentioned this elsewhere, but I think a good UI would be to have a list of visited buffers sorted by _recency of appearing in the activity_ (perhaps cleaning up those that have "fallen off the cart").  So `activities-switch-buffer` would list showing buffers of the active activity first, then the most recently displayed (in this activity) non-showing buffer, etc.

**Reply by alphapapa** [reply]
> I mentioned this elsewhere, but I think a good UI would be to have a list of visited buffers sorted by _recency of appearing in the activity_ (perhaps cleaning up those that have "fallen off the cart"). So `activities-switch-buffer` would list showing buffers of the active activity first, then the most recently displayed (in this activity) non-showing buffer, etc.

It seems to me that it already works basically that way: when I use `activities-switch-buffer`, I see the ones I've switched to most recently within the activity.  But if someone wants to enhance the sorting logic or something like that, I don't object to the idea.

**Reply by jdtsmith** [reply]
Should have added: a _persistent_ list (saved with the last-visited-in-activity timestamp).

**Comment by idlip** [comment]
Tab line is really good, so the design can mean, use tab-bar as workspace/activities, and tab-line kinda as the tabs! But tab-line does not play well with window splitting, as tab-bar does.

This makes more sense, and many would feel right about it I think.
This adds some visual help as well, and makes good use of tab-line!

**Reply by idlip** [reply]
Also I agree that using `recentf` kinda solves and works for what I was expecting. But there is that little hassle.

Nevertheless, gets the work in place enough!

So this is now a feature request if its possible in simple way and benefits others as well in their workflow.

**Reply by alphapapa** [reply]
> Tab line is really good, so the design can mean, use tab-bar as workspace/activities, and tab-line kinda as the tabs! But tab-line does not play well with window splitting, as tab-bar does.

There seems to be a minor misunderstanding here.  I merely suggest that a simple solution (from a UI perspective) to the problem you've described would be to load the buffers in a window's tab line when restoring an activity.  It would mean that we would not have to develop a bespoke UI to manage such buffers.

> Also I agree that using recentf kinda solves and works for what I was expecting. But there is that little hassle.

Again, a small misunderstanding.  I don't propose to use the `recentf` library.  Rather, each window has a list of buffers that have been displayed in it, which are shown in the tab line.

> So this is now a feature request if its possible in simple way and benefits others as well in their workflow.

We'll see.  This idea is already in the notes file: https://github.com/alphapapa/activities.el/blob/meta/notes/NOTES.org#b-optionally-saverestore-tab-line-buffers

**Reply by idlip** [reply]
Oh yes.

Thank you very much for considering this request!

**Reply by alphapapa** [reply]
Filed at https://github.com/alphapapa/activities.el/issues/117
